### ISSUE TITLE
[Bug]: controlnet does not crop to masked inpaint area in img2img

### ISSUE THREAD

**Issue by ride5k** [Mikubill/sd-webui-controlnet#2403]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

img2img inpaint mode
"only masked" inpaint area
create mask which is smaller than original image
enable controlnet (openpose for example, default settings, "pixel perfect" flag set
enter prompt, generate image
controlnet output is generated for entire image, not just inpaint rectangle

this has previously worked flawlessly

### Steps to reproduce the problem

1. Go to img2img inpaint
2. Press enable controlnet, pixel perfect
3. Mask portion of original
4. Generate image which uses an uncropped version of original for controlnet

### What should have happened?

controlnet should crop to inpaint area boundaries, not entire image

### Commit where the problem happens

webui: version: [1.7.0]
python: 3.10.6
torch: 2.0.0+cpu
xformers: N/A
gradio: 3.41.2
controlnet: ControlNet v1.1.427

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--no-half --precision full --no-half-vae --theme=dark --disable-nan-check --disable-safe-unpickle --medvram --sub-quad-q-chunk-size 128 --sub-quad-kv-chunk-size 128 --sub-quad-chunk-threshold 80 --skip-torch-cuda-test --use-directml
```

### List of enabled extensions

built in (ldsr/lora/scunet/swinir/canvas-zoom-and-pan/extra-options-section/hypertile/mobile/prompt-bracket-checker)
only additional extension is sd-webui-controlnet (disabled all others for testing)

### Console logs

```Shell
venv "T:\auto1111\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.7.0
Commit hash: cfa6e40e6d7e290b52940253bf705f282477b890
Installing sd-webui-controlnet requirement: trimesh[easy]
Launching Web UI with arguments: --no-half --precision full --no-half-vae --theme=dark --disable-nan-check --disable-safe-unpickle --medvram --sub-quad-q-chunk-size 128 --sub-quad-kv-chunk-size 128 --sub-quad-chunk-threshold 80 --skip-torch-cuda-test --use-directml
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
ControlNet preprocessor location: T:\auto1111\stable-diffusion-webui-directml\extensions\sd-webui-controlnet\annotator\downloads
2024-01-04 09:38:41,836 - ControlNet - INFO - ControlNet v1.1.427
2024-01-04 09:38:41,917 - ControlNet - INFO - ControlNet v1.1.427
Loading weights [5c8529319b] from T:\auto1111\stable-diffusion-webui-directml\models\Stable-diffusion\inpaint\realisticVisionV60B1_v60B1InpaintingVAE.safetensors
Running on local URL:  http://127.0.0.1:7860
Creating model from config: T:\auto1111\stable-diffusion-webui-directml\configs\v1-inpainting-inference.yaml

To create a public link, set `share=True` in `launch()`.
Startup time: 12.6s (prepare environment: 3.4s, import torch: 3.5s, import gradio: 0.8s, setup paths: 0.6s, initialize shared: 0.9s, other imports: 0.3s, list SD models: 0.4s, load scripts: 1.3s, create ui: 0.7s, gradio launch: 0.7s).
Loading VAE weights specified in settings: T:\auto1111\stable-diffusion-webui-directml\models\VAE\vae-ft-ema-560000-ema-pruned.ckpt
Applying attention optimization: V1... done.
Model loaded in 2.7s (load weights from disk: 0.9s, create model: 0.5s, apply weights to model: 0.7s, load VAE: 0.2s, calculate empty prompt: 0.4s).
2024-01-04 09:38:57,312 - ControlNet - INFO - unit_separate = False, style_align = False
2024-01-04 09:38:57,524 - ControlNet - INFO - Loading model: control_v11p_sd15_openpose [cab727d4]
2024-01-04 09:38:57,952 - ControlNet - INFO - Loaded state_dict from [T:\auto1111\stable-diffusion-webui-directml\extensions\sd-webui-controlnet\models\control_v11p_sd15_openpose.pth]
2024-01-04 09:38:57,953 - ControlNet - INFO - controlnet_default_config
2024-01-04 09:38:59,563 - ControlNet - INFO - ControlNet model control_v11p_sd15_openpose [cab727d4] loaded.
2024-01-04 09:39:00,153 - ControlNet - INFO - Loading preprocessor: openpose_full
2024-01-04 09:39:00,154 - ControlNet - INFO - preprocessor resolution = 512
2024-01-04 09:39:04,154 - ControlNet - INFO - ControlNet Hooked - Time = 6.8450000286102295
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:25<00:00,  1.29s/it]
```

### Additional information

using direct-ml fork of automatic on AMD
